lib/fog/vsphere/models/compute/server.rb

```diff
@@ -348,9 +359,25 @@ def initialize_interfaces
           end
         end
 
+        def unassigned_volumes?
+          attributes[:volumes]&.map! do |vol|
+            return true unless vol.key?(:controller_key)
+          end
+          false
```

Comment:
I wonder why you're using `map!`. Wouldn't that potentially wipe data? Isn't this what you're looking for?
```suggestion
          attributes[:volumes]&.any? { |vol| !vol.key?(:controller_key) } || false
```

Technically `|| false` isn't needed since `nil` and `false` evaluate to the same, but it ensures a proper boolean is returned.

Reply:
that's a much better approach, Thanks!

Reply:
I recommend reading https://docs.ruby-lang.org/en/master/Enumerable.html because it's extremely useful to know. Many things include `Enumerable`, making your life easier.